### ISSUE TITLE
MD037 triggers when using inline LaTeX

### ISSUE THREAD

**Issue by marek-zeleny** [DavidAnson/markdownlint#391]
Hi, there's a very similar problem with `*` within a LaTeX block.
```
Some text $a * b$ some *highlighted* text.
```
Text between the first `*` (within the LaTeX expression) and the second `*` (starting the highlighted text) gets underlined as an error MD037.

By the way, thanks for doing this, it's an awesome tool :)

_Originally posted by @marek-zeleny in https://github.com/DavidAnson/markdownlint/issues/357#issuecomment-826346603_

**Comment by DavidAnson** [comment]
If it's recognized as a code block, that rule should ignore the content like so: https://dlaa.me/markdownlint/#%25m%23%20Issue%20391%0A%0A%60Some%20text%20%24a%20*%20b%24%20some%20*highlighted*%20text.%60%0A

But by default math blocks are not recognized by the parser so it's probably treating this like text. You can add a plug-in for math blocks and it should fix this problem by ignoring them. Here's an example test case that does that with the `markdown-it-texmath` plugin: https://github.com/DavidAnson/markdownlint/blob/main/test/markdownlint-test.js#L1459-L1481

**Comment by marek-zeleny** [comment]
I was using a math parser included in [Markdown All in One](https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one) and after reading your suggestion I tried using [Markdown+Math](https://marketplace.visualstudio.com/items?itemName=goessner.mdmath) instead, but neither of them helped resolving the issue. `Markdown+Math` even claims to be built on `markdown-it-texmath`. Do I have to explicitely use the plugin within `markdownlint` (which would be quite inconvenient), or is there some other problem in my approach?

Thanks

**Comment by DavidAnson** [comment]
How are you doing this all? Are you using the VS Code extension for `markdownlint`? It includes `markdown-it-texmath` by default. Are you using `markdownlint-cli2`? That lets you reference plugins. Please describe your scenario.

**Comment by marek-zeleny** [comment]
I'm using VS Code with extensions `markdownlint`, `Markdown All in One` and now I tried also `Markdown+Math`. I haven't been using `markdownlint-cli2` yet, because I'm mainly on Windows and if I understand it correctly, `cli2` can be used (apart from CLI) only as a project-specific file, while I'd like to have this as a global setting.

**Comment by DavidAnson** [comment]
Please share a simple example of Markdown that is flagging this error unexpectedly. I will transfer this issue to that project if applicable.

I don't think the other extensions in your scenario here are helping or hurting for this specific issue. CLI2 expects to be configured per project, yes.

**Comment by lihykang** [comment]
Tôi không nhận được máy

Được gửi từ Thư<https://go.microsoft.com/fwlink/?LinkId=550986> dành cho Windows 10

Từ: David ***@***.***>
Đã gửi: 26 Tháng Tư 2021 6:48 SA
Đến: ***@***.***>
Cc: ***@***.***>
Chủ đề: Re: [DavidAnson/markdownlint] MD037 triggers when using inline LaTeX (#391)


Please share a simple example of Markdown that is flagging this error unexpectedly. I will transfer this issue to that project if applicable.

I don't think the other extensions in your scenario here are helping or hurting for this specific issue. CLI2 expects to be configured per project, yes.

**Comment by marek-zeleny** [comment]
I tried adding the `markdown-it-texmath` plugin in `markdownlint-cli2` and the previous error now disappeared, so it probably works as expected. I was using the example I gave in the opening comment of this ticket.

Thank you for your very quick and helpful responses. If you like, you might take this as a suggestion - it would be nice to have some global settings of the activated plugins (and possibly other things configurable by `cli2`) with the possibility to override them locally per project.

Have a nice day and thank you again for this project, it's really great!
